pkg/logs/tail.go

```diff
@@ -0,0 +1,603 @@
+				// for this RequestID
+				highSeq := highSequence(logs)
+
+				// Whether we have the RequestID or not, we
+				// append and sort the logs slice.
+				reqLogs := logmap[req]
+				reqLogs.logs = append(reqLogs.logs, logs...)
+				// Sort the current batch of logs by their sequence number.
+				sort.Slice(reqLogs.logs,
+					func(i, j int) bool {
+						return reqLogs.logs[i].SequenceNum < reqLogs.logs[j].SequenceNum
+					})
+
+				// Check to see if we already have a timer
+				// running or if the current high sequence is
+				// higher than the one with the timer.
+				// The timer will always be running on the head
+				// of the slice.
+				recv := reqLogs.receives
+
+				// In either case append to the receives slice.
+				if len(recv) == 0 || recv[0].highSeq < highSeq {
+					reqLogs.receives = append(recv, receive{
+						when:    time.Now(),
+						highSeq: highSeq,
+					})
+				}
+
+				// In only the empty case, start a new timer
+				// since this is the head of the slice.
+				if len(recv) == 0 {
+					time.AfterFunc(c.cfg.sortBuffer, func() {
+						tdCh <- bufferedLog{
+							reqID: req,
+							seq:   highSeq,
+						}
+					})
+				}
+
+				// Set the new log and receive info back to the
+				// logmap for this RequestID.
+				logmap[req] = reqLogs
+			}
+
+		case bufdLogs := <-tdCh: // A timer expired for a particular request.
+			reqID, seq := bufdLogs.reqID, bufdLogs.seq
+
+			// Get the logs for this RequestID and
+			// find the index of the sequence in our current logs.
+			reqLogs := logmap[reqID]
+			idx := findIdxBySeq(reqLogs.logs, seq)
+
+			// Split off the source of this timer, leave
+			// remaining logs to be printed later.
+			toPrint, remainingLogs := reqLogs.logs[:idx], reqLogs.logs[idx:]
+			reqLogs.logs = remainingLogs
+			c.printLogs(out, toPrint)
+
+			// Special case if we just printed the entire set of
+			// logs, we remove the keys from the maps and finish.
+			if len(remainingLogs) == 0 {
+				delete(logmap, reqID)
+				break
+			}
+
+			// Drop the front of the batchReqReceives map and start
+			// another timer for any remaining recorded sequences.
+			recv := reqLogs.receives[1:]
+			reqLogs.receives = recv
+
+			// If anything is left...
+			if len(recv) > 0 {
+				// We create a new timer, we subtract
+				// off time already served from the
+				// user defined sortBuffer.
+				time.AfterFunc(c.cfg.sortBuffer-time.Since(recv[0].when), func() {
+					tdCh <- bufferedLog{
+						reqID: reqID,
+						seq:   recv[0].highSeq,
+					}
+				})
+			}
+
+			// Set the new log and receive info back to the
+			// logmap for this RequestID.
+			logmap[reqID] = reqLogs
+
+		case <-c.doneCh:
+			os.Exit(0)
+		}
+	}
+}
+
+// printLogs is a simple printer for Log slices, only printing requested
+// streams.
+func (c *TailCommand) printLogs(out io.Writer, logs []Log) {
+	if len(logs) > 0 {
+		filtered := filterStream(c.cfg.stream, logs)
+
+		for _, l := range filtered {
+			text.Output(out, l.String())
```

Comment:
The `text.Output` function internally 'wraps' the output where as this sort of data should flow as wide on the screen as necessary otherwise users will find that piping the data to another command will break.

It was suggested by Patrick that there isn't another function within the `text` package we could use that didn't also wrap the text output so we should look to replace `text.Output` with a standard `fmt.Print<T>` variation.

Reply:
changed to fprintln